### PR TITLE
Ensure `fit` reproducibility

### LINKED ISSUE #958: Reproducibility Issues in mmm.fit()
Hi there! I seem to be unable to reproduce my MMMs across runs. Similarly, I'm unable to reproduce the results obtained in the PyMC-Marketing example notebook. 

Any advice on how to resolve this would be greatly appreciated! 🙏 

It seems like my rng is not being applied correctly when I call mmm.fit().

Evidence in screenshots, notice e.g. how adstock_alpha changes:
image
image

```
import pymc as pm
import pymc_marketing
print(pm.__version__)
print(pymc_marketing.__version__)

# Create MMM with pymc-marketing
mmm = MMM(
    date_column=c.DATE_COLUMN,
    channel_columns=FEATURES["ft_channels"],
    control_columns=FEATURES["ft_control"] + ["trend"],
    adstock="geometric",
    saturation="logistic",
    adstock_max_lag=4,
    sampler_config={"progressbar": False},
    validate_data=False, # Necessary to allow negative competitor spend
)


# Run MMM with the experiment results included
rng: np.random.Generator = np.random.default_rng(seed=42)
idata = mmm.fit(X_train, y_train, random_seed=rng)

idata.posterior.adstock_alpha.mean()


# Create MMM with pymc-marketing
mmm = MMM(
    date_column=c.DATE_COLUMN,
    channel_columns=FEATURES["ft_channels"],
    control_columns=FEATURES["ft_control"] + ["trend"],
    adstock="geometric",
    saturation="logistic",
    adstock_max_lag=4,
    sampler_config={"progressbar": False},
    validate_data=False, # Necessary to allow negative competitor spend
)


# Run MMM with the experiment results included
rng: np.random.Generator = np.random.default_rng(seed=42)
idata2 = mmm.fit(X_train, y_train, random_seed=rng)

idata2.posterior.adstock_alpha.mean()
```

cc @iraur

### LINKED ISSUE #961: Ensure same results with `ModelBuilder.fit` random_seed
Brought up in #958, there are different results depending on the way on the random_seed is set.

```python
# Doesn't work
model.fit(random_seed=42)
model.fit(..., random_seed=np.random.default_rng(42))
# Does work
model = ModelBuilder(..., sampler_config={"random_seed": 42})
model.fit(...)
# Doesn't work because of np.random.Generator is not serializable
model = ModelBuilder(..., sampler_config={"random_seed": np.random.default_rng(42)})
model.fit(...)
```

Current workaround: Use an int `random_seed` parameter in `sampler_config` at initialization.

See the #958 for more details.

### PR DESCRIPTION
**Pull request by wd60622**
## Related Issue

- [ ] Closes #961 Closes #958
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--963.org.readthedocs.build/en/963/

### CODE REVIEW

**Comment by wd60622** [comment]
@louismagowan @iraur

File: pymc_marketing/model_builder.py [hunk 1]
```diff
@@ -516,8 +556,8 @@ def fit(
             The training input samples. If scikit-learn is available, array-like, otherwise array.
         y : array-like | array, shape (n_obs,)
             The target values (real numbers). If scikit-learn is available, array-like, otherwise array.
-        progressbar : bool
-            Specifies whether the fit progress bar should be displayed.
+        progressbar : bool, optional
+            Specifies whether the fit progress bar should be displayed. Defaults to True.
```
Comment (juanitorduz) [inline pymc_marketing/model_builder.py:560]: In the docstrings, it says it defaults to `None` . Could you please clarify this :) ?

**Review by juanitorduz — APPROVED** [review]
Thanks Will! Just a small comment to fix in the docstring :)

File: pymc_marketing/model_builder.py (hunk 1 above)
Comment (wd60622) [reply pymc_marketing/model_builder.py:560]: I have "Defaults to True" in the description. What else did you have in mind?

**Pull request merged**